xapian-core/cluster/kmeans.cc

```diff
@@ -24,120 +24,126 @@
 #include "xapian/cluster.h"
 #include <debuglog.h>
 #include <xapian/error.h>
+#include <omassert.h>
 
-#include <time.h>
-#include <cstdlib>
 #include <climits>
+#include <random>
 #include <set>
 #include <unordered_map>
 #include <vector>
 
+/** Threshold value for checking convergence in KMeans
+ */
 #define EPSILON 0.0000000001
 
 using namespace Xapian;
 using namespace std;
 
+KMeans::KMeans(unsigned int k_) {
+    LOGCALL_CTOR(API, "KMeans()", k_);
+    if (k == 0)
+	throw InvalidArgumentError("Number of required clusters should be greater than zero");
+    k = k_;
+    max_iters = 100;
+}
+
+KMeans::KMeans(unsigned int k_, unsigned int max_iters_) {
+    LOGCALL_CTOR(API, "KMeans()", k_ | max_iters_);
+    if (k == 0)
+	throw InvalidArgumentError("Number of required clusters should be greater than zero");
+    k = k_;
+    max_iters = max_iters_;
+}
```

Comment:
These two can be simplified if you give `max_iters_` a default value. (Also, you can use an initialiser list to set `max_iters`.)